sdk/core/core/inc/az_http_request_builder.h

```diff
@@ -12,26 +12,40 @@
 #include <az_result.h>
 #include <az_span.h>
 
+#include <stdint.h>
+
 #include <_az_cfg_prefix.h>
 
 typedef struct {
   az_span buffer;
-  int16_t max_headers;
-  int16_t max_url_size;
   az_const_span method_verb;
-  int16_t retry_headers_start;
-  int16_t headers_end;
+  az_span url;
+  uint16_t max_url_size;
+  uint16_t max_headers;
+  uint16_t retry_headers_start;
+  uint16_t headers_end;
 } az_http_request_builder;
 
+extern az_const_span const AZ_HTTP_METHOD_VERB_GET;
+extern az_const_span const AZ_HTTP_METHOD_VERB_HEAD;
+extern az_const_span const AZ_HTTP_METHOD_VERB_POST;
+extern az_const_span const AZ_HTTP_METHOD_VERB_PUT;
+extern az_const_span const AZ_HTTP_METHOD_VERB_DELETE;
+extern az_const_span const AZ_HTTP_METHOD_VERB_TRACE;
+extern az_const_span const AZ_HTTP_METHOD_VERB_OPTIONS;
+extern az_const_span const AZ_HTTP_METHOD_VERB_CONNECT;
+extern az_const_span const AZ_HTTP_METHOD_VERB_PATCH;
+
 /**
- * @brief format buffer as a http request containing headers, url and body.
+ * @brief format buffer as a http request containing headers and url.
  */
 az_result az_http_request_builder_init(
     az_http_request_builder * const p_hrb,
     az_span const buffer,
-    int16_t const max_url_size,
     az_const_span const method_verb,
-    az_const_span const initial_url);
+    az_const_span const initial_url,
+    uint16_t const max_url_size,
+    uint16_t const max_headers);
```

Comment:
I think the parameter order should be: buffer, max_url_size, method_verb, initial_url
The "configuration" of the buffer is set by buffer & max_url_size
The initial values in the configured buffer are set by method_verb & initial_url
The max_headers is NOT passed by the user but is instead calculated by this function based on the buffer size - max_url_size bytes divided by the az_pair structure size (also you have to taking alignment into affect)